forest/willow/log_stats.py

```diff
@@ -437,121 +437,104 @@ def log_stats_main(
-                            "num_out_caller", "num_mis_caller"
-                        ]:
-                            sum_all_set_cols += stats_pdframe[col]
-                            if (
-                                stats_pdframe[
-                                    "num_uniq_individuals_call_or_text"
-                                ] < stats_pdframe[col]
-                            ).any():
-                                logger.error(
-                                    "Error: "
-                                    "num_uniq_individuals_call_or_text "
-                                    "was found to be less than %s for at "
-                                    "least one time interval. This error "
-                                    "comes from an issue with the code,"
-                                    " not an issue with the input data",
-                                    col
-                                    )
-                        if (
-                            stats_pdframe[
-                                "num_uniq_individuals_call_or_text"
-                            ] > sum_all_set_cols
-                        ).any():
-                            logger.error(
-                                    "Error: "
-                                    "num_uniq_individuals_call_or_text "
-                                    "was found to be larger than the sum "
-                                    "of individual cardinalities for at "
-                                    "least one time interval. This error "
-                                    "comes from an issue with the code,"
-                                    " not an issue with the input data"
-                                    )
-
-                        write_all_summaries(bid, stats_pdframe, output_folder)
-
-                    logger.info(
-                        "Summary statistics obtained. Finished."
-                    )
+    logger.info("Summary statistics obtained. Finished.")
 
-            except Exception as err:
-                logger.error(
-                    "An error occurred when processing the data: %s", err
-                )
+
+def log_stats_inner(
+    beiwe_id: str,
+    output_folder: str,
+    study_folder: str,
+    frequency: Frequency,
+    tz_str: str,
+    time_start: Optional[List] = None,
+    time_end: Optional[List] = None,
+):
+    """ Inner functionality of log_stats_main """
+    # read data
+    text_data, text_stamp_start, text_stamp_end = read_data(
+        beiwe_id, study_folder, "texts", tz_str, time_start, time_end
+    )
+    call_data, call_stamp_start, call_stamp_end = read_data(
+        beiwe_id, study_folder, "calls", tz_str, time_start, time_end
+    )
+
+    # give up early if there is no data
+    if text_data.shape[0] <= 0 and call_data.shape[0] <= 0:
+        logger.info(f"There was no data for participant {beiwe_id}")
```

Comment:
We use `logger.info("%s", var)` style in log strings (https://forest.beiwe.org/en/latest/development.html#python)